src/main/java/org/apache/pulsar/io/jcloud/partitioner/AbstractPartitioner.java

```diff
@@ -42,35 +46,67 @@ public abstract class AbstractPartitioner<T> implements Partitioner<T> {
+    }
+
+    private Map<String, String> parseTopicsToPathMappingString(String topicsToPathMappingString) {
+        Map<String, String> topicsToPathMapping = new HashMap<>();
+        if (StringUtils.isNotEmpty(topicsToPathMappingString)) {
+            for (String topicPathPair : topicsToPathMappingString.split(",")) {
+                String[] topicToPathMapping = topicPathPair.split("=");
+                if (topicToPathMapping.length == 2) {
+                    topicsToPathMapping.put(topicToPathMapping[0], topicToPathMapping[1]);
+                } else {
+                    LOGGER.warn("The topic to path mapping is not in correct format {}.", topicPathPair);
+                }
+            }
+        }
+        LOGGER.info("The final topics to path mapping is: {}", topicsToPathMapping);
+        return topicsToPathMapping;
     }
 
     @Override
     public String generatePartitionedPath(String topic, String encodedPartition) {
+        if (MapUtils.isNotEmpty(computedTopicsToPathMapping) && computedTopicsToPathMapping.containsKey(topic)) {
+            return StringUtils.join(Arrays.asList(computedTopicsToPathMapping.get(topic), encodedPartition),
+                    PATH_SEPARATOR);
+        } else {
+            List<String> joinList = new ArrayList<>();
+            TopicName topicName = TopicName.get(topic);
+            joinList.add(topicName.getTenant());
+            joinList.add(topicName.getNamespacePortion());
 
-        List<String> joinList = new ArrayList<>();
-        TopicName topicName = TopicName.get(topic);
-        joinList.add(topicName.getTenant());
-        joinList.add(topicName.getNamespacePortion());
-
-        if (topicName.isPartitioned() && withTopicPartitionNumber) {
-            if (sliceTopicPartitionPath) {
+            if (topicName.isPartitioned() && withTopicPartitionNumber) {
+                if (sliceTopicPartitionPath) {
+                    TopicName newTopicName = TopicName.get(topicName.getPartitionedTopicName());
+                    joinList.add(newTopicName.getLocalName());
+                    joinList.add(Integer.toString(topicName.getPartitionIndex()));
+                } else {
+                    joinList.add(topicName.getLocalName());
+                }
+            } else {
                 TopicName newTopicName = TopicName.get(topicName.getPartitionedTopicName());
                 joinList.add(newTopicName.getLocalName());
-                joinList.add(Integer.toString(topicName.getPartitionIndex()));
-            } else {
-                joinList.add(topicName.getLocalName());
             }
-        } else {
-            TopicName newTopicName = TopicName.get(topicName.getPartitionedTopicName());
-            joinList.add(newTopicName.getLocalName());
+            String generatedTopicPath = StringUtils.join(joinList, PATH_SEPARATOR);
+            if (MapUtils.isNotEmpty(topicsToPathMapping) && topicsToPathMapping.containsKey(generatedTopicPath)) {
```

Comment:
no need to check if empty, please remove

Reply:
removed